google/cloud/sql/integration_tests/sql_integration_test.cc

```diff
@@ -97,21 +99,30 @@ TEST_F(SqlIntegrationTest, CreateEphemeral) {
   ASSERT_NE(instances.begin(), instances.end());
   ASSERT_STATUS_OK(*instances.begin());
 
-  google::cloud::sql::v1::SqlInstancesCreateEphemeralCertRequest request;
-  request.set_project(project_id_);
-  request.set_instance((*instances.begin())->name());
-  request.mutable_body()->set_public_key("THE_PUBLIC_KEY");
+  // All the RPCs that could repro #12112 are non-idempotent. So we put a little
+  // manual retry loop around the test.
+  auto const attempt_delay = std::chrono::seconds(30);
+  for (int attempt = 0; attempt != 3; ++attempt) {
+    if (attempt != 0) std::this_thread::sleep_for(attempt_delay);
+    google::cloud::sql::v1::SqlInstancesCreateEphemeralCertRequest request;
+    request.set_project(project_id_);
+    request.set_instance((*instances.begin())->name());
+    request.mutable_body()->set_public_key("THE_PUBLIC_KEY");
```

Comment:
It looks like `request` initialization could be hoisted from the loop.

Reply:
Done.